examples/with-apollo/lib/withData.js

```diff
@@ -0,0 +1,49 @@
+import { ApolloProvider, getDataFromTree } from 'react-apollo'
+import React from 'react'
+import 'isomorphic-fetch'
+import { initClient } from './initClient'
+import { initStore } from './initStore'
+
+export default (Component) => (
+  class extends React.Component {
+    static async getInitialProps (ctx) {
+      const headers = ctx.req ? ctx.req.headers : {}
+      const client = initClient(headers)
+      const store = initStore(client, client.initialState)
+
+      if (!process.browser) {
+        const app = (
+          <ApolloProvider client={client} store={store}>
+            <Component url={{ query: ctx.query, pathname: ctx.pathname }} />
```

Comment:
Good spot @sedubois.  I just pushed this up to my fork. I inlined the conditional rather then create a new util file. You can check it out here: https://github.com/ads1018/next.js/commit/30de6979bc5927bd62f6b55c0cf4be47100868e6

I can create another pull request for this.